### ISSUE TITLE
Harden System.Net.Http.Functional.Tests.HttpClientHandler_SslProtocols_Test.GetAsync_SupportedSSLVersion_Succeeds and GetAsync_UnsupportedSSLVersion_Throws

### ISSUE THREAD

**Issue by steveharter** [dotnet/corefx#14767]
These two tests account for 78 CI failures from Dec 1 - 28 which is 10% of System.Net* failures.

Assuming the tests fail because they access www.ssllabs.com which (apparently) is not reliable. However, since the failures are coming mostly from centos that needs to be taken into consideration as a cause as well.

If the failures are due to www.ssllabs.com being unreliable, we should add a check to ensure the web site is available before invoking it for the test. Or come up with some other similar mechanism to reduce CI failures.

````
Test Case: System.Net.Http.Functional.Tests.HttpClientHandler_SslProtocols_Test.GetAsync_SupportedSSLVersion_Succeeds(name: \"TLSv1.2\", url: \"https://www.ssllabs.com:10303/\")
Failed Jenkins Jobs
Build Number 	Machine Name 	Date
dotnet_corefx/master/outerloop_centos7.1_release 28 	centos71-20160609-1-outer39e350 	12/01 04:32 PM
dotnet_corefx/master/outerloop_centos7.1_debug 29 	centos71-20160609-1-outer5140a0 	12/01 07:53 AM
dotnet_corefx/master/outerloop_centos7.1_release 29 	centos71-20160609-1-outerf7e7a0 	12/02 04:33 PM
dotnet_corefx/master/outerloop_centos7.1_debug 30 	centos71-20160609-1-outer8e8da0 	12/02 07:53 AM
dotnet_corefx/master/outerloop_centos7.1_release 30 	centos71-20160609-1-outerec3110 	12/03 04:32 PM
dotnet_corefx/master/outerloop_centos7.1_release 31 	centos71-20160609-1-outere07ae0 	12/04 04:33 PM
dotnet_corefx/master/outerloop_centos7.1_debug 32 	centos71-20160609-1-outer772080 	12/04 07:52 AM
dotnet_corefx/master/outerloop_centos7.1_debug 33 	centos71-20160609-1-outer6b69f0 	12/05 07:52 AM
dotnet_corefx/master/outerloop_centos7.1_release 32 	centos71-20160609-1-outerc90b80 	12/06 04:32 PM
dotnet_corefx/master/outerloop_centos7.1_release 33 	centos71-20160609-1-outerbd57a0 	12/07 04:32 PM
dotnet_corefx/master/outerloop_centos7.1_debug 34 	centos71-20160609-1-outer53fda0 	12/07 07:52 AM
dotnet_corefx/master/outerloop_centos7.1_release 34 	centos71-20160609-1-outerb19300 	12/08 04:32 PM
dotnet_corefx/master/outerloop_centos7.1_debug 35 	centos71-20160609-1-outer484710 	12/08 07:52 AM
dotnet_corefx/master/outerloop_opensuse13.2_debug 35 	suse132-20160315-outer6a28c0 	12/08 05:14 PM
dotnet_corefx/dev_apple_crypto/outerloop_centos7.1_debug 4 	centos71-20160609-1-outer43c870 	12/08 09:35 PM
dotnet_corefx/master/outerloop_centos7.1_release 35 	centos71-20160609-1-outera5ea80 	12/09 04:32 PM
dotnet_corefx/master/outerloop_centos7.1_release 36 	centos71-20160609-1-outera7cb80 	12/10 04:35 PM
dotnet_corefx/master/outerloop_centos7.1_debug 37 	centos71-20160609-1-outer30d750 	12/10 07:52 AM
dotnet_corefx/master/outerloop_centos7.1_release 37 	centos71-20160609-1-outer9c2310 	12/11 04:34 PM
dotnet_corefx/master/outerloop_centos7.1_debug 38 	centos71-20160609-1-outer2f2700 	12/11 07:52 AM
dotnet_corefx/master/outerloop_centos7.1_release 38 	centos71-20160609-1-outer8deec0 	12/12 12:19 PM
dotnet_corefx/master/outerloop_centos7.1_debug 39 	centos71-20160609-1-outer237070 	12/12 07:52 AM
dotnet_corefx/master/outerloop_centos7.1_debug 40 	centos71-20160609-1-outer86f110 	12/12 12:14 PM
dotnet_corefx/master/outerloop_centos7.1_release 39 	centos71-20160609-1-outer84a7e0 	12/13 04:35 PM
dotnet_corefx/master/outerloop_centos7.1_debug 41 	centos71-20160609-1-outer5df8f0 	12/13 07:52 AM
dotnet_corefx/master/outerloop_centos7.1_release 41 	centos71-20160609-1-outer3162b0 	12/15 04:30 PM
dotnet_corefx/master/outerloop_centos7.1_debug 43 	centos71-20160609-1-outer4ccf70 	12/15 07:52 AM
dotnet_corefx/master/outerloop_opensuse42.1_release 42 	suse421-20160920-outer71c900 	12/16 12:46 PM
dotnet_corefx/master/outerloop_centos7.1_debug 44 	centos71-20160609-1-outerf495c0 	12/16 07:52 AM
dotnet_corefx/master/outerloop_centos7.1_release 43 	centos71-20160609-1-outer578200 	12/17 04:35 PM
dotnet_corefx/master/outerloop_centos7.1_debug 45 	centos71-20160609-1-outere8df40 	12/17 07:52 AM
dotnet_corefx/master/outerloop_centos7.1_release 44 	centos71-20160609-1-outer4a24b0 	12/18 04:34 PM
dotnet_corefx/master/outerloop_centos7.1_debug 46 	centos71-20160609-1-outere26360 	12/18 07:56 AM
dotnet_corefx/master/outerloop_centos7.1_release 45 	centos71-20160609-1-outer7e37d0 	12/20 04:30 PM
dotnet_corefx/master/outerloop_centos7.1_debug 47 	centos71-20160609-1-outercaf640 	12/20 07:56 AM
dotnet_corefx/master/outerloop_centos7.1_release 46 	centos71-20160609-1-outer2d8f90 	12/21 04:32 PM
dotnet_corefx/master/outerloop_centos7.1_debug 49 	centos71-20160609-1-outerb854b0 	12/22 07:52 AM
dotnet_corefx/dev_apple_crypto/outerloop_centos7.1_debug 5 	centos71-20160609-1-outerb04220 	12/22 09:33 PM
dotnet_corefx/dev_apple_crypto/outerloop_centos7.1_release 5 	centos71-20160609-1-outere5a090 	12/22 05:19 AM
dotnet_corefx/master/outerloop_centos7.1_release 48 	centos71-20160609-1-outer199a30 	12/23 04:35 PM
dotnet_corefx/master/outerloop_centos7.1_debug 50 	centos71-20160609-1-outerb75f40 	12/23 07:53 AM
dotnet_corefx/master/outerloop_centos7.1_debug 51 	centos71-20160609-1-outer897c50 	12/26 07:52 AM
dotnet_corefx/master/outerloop_centos7.1_debug 52 	centos71-20160609-1-outer720f40 	12/28 07:53 AM
````
````
Test Case: System.Net.Http.Functional.Tests.HttpClientHandler_SslProtocols_Test.GetAsync_SupportedSSLVersion_Succeeds(name: \"TLSv1.1\", url: \"https://www.ssllabs.com:10302/\")
Failed Jenkins Jobs
Build Number 	Machine Name 	Date
dotnet_corefx/master/outerloop_centos7.1_release 28 	centos71-20160609-1-outer39e350 	12/01 04:32 PM
dotnet_corefx/master/outerloop_centos7.1_debug 29 	centos71-20160609-1-outer5140a0 	12/01 07:53 AM
dotnet_corefx/master/outerloop_centos7.1_release 29 	centos71-20160609-1-outerf7e7a0 	12/02 04:33 PM
dotnet_corefx/master/outerloop_centos7.1_debug 30 	centos71-20160609-1-outer8e8da0 	12/02 07:53 AM
dotnet_corefx/master/outerloop_centos7.1_release 30 	centos71-20160609-1-outerec3110 	12/03 04:32 PM
dotnet_corefx/master/outerloop_centos7.1_release 31 	centos71-20160609-1-outere07ae0 	12/04 04:33 PM
dotnet_corefx/master/outerloop_centos7.1_debug 32 	centos71-20160609-1-outer772080 	12/04 07:52 AM
dotnet_corefx/master/outerloop_centos7.1_debug 33 	centos71-20160609-1-outer6b69f0 	12/05 07:52 AM
dotnet_corefx/master/outerloop_centos7.1_release 32 	centos71-20160609-1-outerc90b80 	12/06 04:32 PM
dotnet_corefx/master/outerloop_centos7.1_release 33 	centos71-20160609-1-outerbd57a0 	12/07 04:32 PM
dotnet_corefx/master/outerloop_centos7.1_debug 34 	centos71-20160609-1-outer53fda0 	12/07 07:52 AM
dotnet_corefx/master/outerloop_centos7.1_release 34 	centos71-20160609-1-outerb19300 	12/08 04:32 PM
dotnet_corefx/master/outerloop_centos7.1_debug 35 	centos71-20160609-1-outer484710 	12/08 07:52 AM
dotnet_corefx/master/outerloop_opensuse13.2_debug 35 	suse132-20160315-outer6a28c0 	12/08 05:14 PM
dotnet_corefx/dev_apple_crypto/outerloop_centos7.1_debug 4 	centos71-20160609-1-outer43c870 	12/08 09:35 PM
dotnet_corefx/master/outerloop_centos7.1_release 35 	centos71-20160609-1-outera5ea80 	12/09 04:32 PM
dotnet_corefx/master/outerloop_centos7.1_release 36 	centos71-20160609-1-outera7cb80 	12/10 04:35 PM
dotnet_corefx/master/outerloop_centos7.1_debug 37 	centos71-20160609-1-outer30d750 	12/10 07:52 AM
dotnet_corefx/master/outerloop_centos7.1_release 37 	centos71-20160609-1-outer9c2310 	12/11 04:34 PM
dotnet_corefx/master/outerloop_centos7.1_debug 38 	centos71-20160609-1-outer2f2700 	12/11 07:52 AM
dotnet_corefx/master/outerloop_centos7.1_release 38 	centos71-20160609-1-outer8deec0 	12/12 12:19 PM
dotnet_corefx/master/outerloop_centos7.1_debug 39 	centos71-20160609-1-outer237070 	12/12 07:52 AM
dotnet_corefx/master/outerloop_centos7.1_debug 40 	centos71-20160609-1-outer86f110 	12/12 12:14 PM
dotnet_corefx/master/outerloop_centos7.1_release 39 	centos71-20160609-1-outer84a7e0 	12/13 04:35 PM
dotnet_corefx/master/outerloop_centos7.1_debug 41 	centos71-20160609-1-outer5df8f0 	12/13 07:52 AM
dotnet_corefx/master/outerloop_centos7.1_release 41 	centos71-20160609-1-outer3162b0 	12/15 04:30 PM
dotnet_corefx/master/outerloop_centos7.1_debug 43 	centos71-20160609-1-outer4ccf70 	12/15 07:52 AM
dotnet_corefx/master/outerloop_centos7.1_debug 44 	centos71-20160609-1-outerf495c0 	12/16 07:52 AM
dotnet_corefx/master/outerloop_centos7.1_release 43 	centos71-20160609-1-outer578200 	12/17 04:35 PM
dotnet_corefx/master/outerloop_centos7.1_debug 45 	centos71-20160609-1-outere8df40 	12/17 07:52 AM
dotnet_corefx/master/outerloop_centos7.1_release 44 	centos71-20160609-1-outer4a24b0 	12/18 04:34 PM
dotnet_corefx/master/outerloop_centos7.1_debug 46 	centos71-20160609-1-outere26360 	12/18 07:56 AM
dotnet_corefx/master/outerloop_centos7.1_release 45 	centos71-20160609-1-outer7e37d0 	12/20 04:30 PM
dotnet_corefx/master/outerloop_centos7.1_debug 47 	centos71-20160609-1-outercaf640 	12/20 07:56 AM
dotnet_corefx/master/outerloop_centos7.1_release 46 	centos71-20160609-1-outer2d8f90 	12/21 04:32 PM
dotnet_corefx/master/outerloop_centos7.1_debug 49 	centos71-20160609-1-outerb854b0 	12/22 07:52 AM
dotnet_corefx/dev_apple_crypto/outerloop_centos7.1_debug 5 	centos71-20160609-1-outerb04220 	12/22 09:33 PM
dotnet_corefx/dev_apple_crypto/outerloop_centos7.1_release 5 	centos71-20160609-1-outere5a090 	12/22 05:19 AM
dotnet_corefx/master/outerloop_centos7.1_release 48 	centos71-20160609-1-outer199a30 	12/23 04:35 PM
dotnet_corefx/master/outerloop_centos7.1_debug 50 	centos71-20160609-1-outerb75f40 	12/23 07:53 AM
dotnet_corefx/master/outerloop_centos7.1_debug 51 	centos71-20160609-1-outer897c50 	12/26 07:52 AM
dotnet_corefx/master/outerloop_centos7.1_debug 52 	centos71-20160609-1-outer720f40 	12/28 07:53 AM
````
````
Test Case: System.Net.Http.Functional.Tests.HttpClientHandler_SslProtocols_Test.GetAsync_SupportedSSLVersion_Succeeds(name: \"TLSv1.0\", url: \"https://www.ssllabs.com:10301/\")
Failed Jenkins Jobs
Build Number 	Machine Name 	Date
dotnet_corefx/master/outerloop_opensuse13.2_debug 35 	suse132-20160315-outer6a28c0 	12/08 05:14 PM
dotnet_corefx/master/outerloop_opensuse42.1_release 42 	suse421-20160920-outer71c900 	12/16 12:46 PM
dotnet_corefx/master/outerloop_rhel7.2_debug 47 	rhel72-20160412-1-outer980a50 	12/17 06:56 PM
````

````
Test Case: System.Net.Http.Functional.Tests.HttpClientHandler_SslProtocols_Test.GetAsync_SupportedSSLVersion_Succeeds(name: "TLSv1.0", url: "https://www.ssllabs.com:10301/")
Failed Jenkins Jobs
Build Number 	Machine Name 	Date
dotnet_corefx/release_1.1.0/outerloop_rhel7.2_release 10 	rhel72-20160412-1-outer8e7030 	12/10 02:45 PM
````

````
est Case: System.Net.Http.Functional.Tests.HttpClientHandler_SslProtocols_Test.GetAsync_UnsupportedSSLVersion_Throws(name: \"SSLv3\", url: \"https://www.ssllabs.com:10300/\")
Failed Jenkins Jobs
Build Number 	Machine Name 	Date
dotnet_corefx/dev_apple_crypto/outerloop_centos7.1_debug 4 	centos71-20160609-1-outer43c870 	12/08 09:35 PM
dotnet_corefx/dev_apple_crypto/outerloop_rhel7.2_debug 4 	rhel72-20160412-1-outer967bf0 	12/09 12:18 AM
dotnet_corefx/dev_apple_crypto/outerloop_rhel7.2_release 4 	rhel72-20160412-1-outerf86c30 	12/09 02:00 PM
dotnet_corefx/dev_apple_crypto/outerloop_centos7.1_debug 5 	centos71-20160609-1-outerb04220 	12/22 09:33 PM
dotnet_corefx/dev_apple_crypto/outerloop_centos7.1_release 5 	centos71-20160609-1-outere5a090 	12/22 05:19 AM
dotnet_corefx/dev_apple_crypto/outerloop_rhel7.2_debug 5 	rhel72-20160412-1-outer124e30 	12/22 12:19 AM
dotnet_corefx/dev_apple_crypto/outerloop_rhel7.2_release 5 	rhel72-20160412-1-outer743060 	12/22 02:00 PM
````

Typical errors:
````
System.Net.Http.Functional.Tests.HttpClientHandler_SslProtocols_Test.GetAsync_SupportedSSLVersion_Succeeds(name: \"TLSv1.0\", url: \"https://www.ssllabs.com:10301/\") (from (empty))
Failing for the past 1 build (Since Failed#47 )
Took 1.4 sec.
Stacktrace

MESSAGE:
System.Net.Http.HttpRequestException : An error occurred while sending the request.\n---- System.Net.Http.CurlException : Problem with the SSL CA cert (path? access rights?)
+++++++++++++++++++
STACK TRACE:
at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult() at System.Net.Http.HttpClient.<FinishSendAsyncBuffered>d__58.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult() at System.Net.Http.Functional.Tests.HttpClientHandler_SslProtocols_Test.<GetAsync_SupportedSSLVersion_Succeeds>d__6.MoveNext() in /mnt/resource/j/workspace/dotnet_corefx/master/outerloop_rhel7.2_debug/src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.SslProtocols.cs:line 127 --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) ----- Inner Stack Trace ----- at System.Net.Http.CurlHandler.ThrowIfCURLEError(CURLcode error) at System.Net.Http.CurlHandler.MultiAgent.FinishRequest(StrongToWeakReference`1 easyWrapper, CURLcode messageResult)
````
````
System.Net.Http.Functional.Tests.HttpClientHandler_SslProtocols_Test.GetAsync_SupportedSSLVersion_Succeeds(name: \"TLSv1.2\", url: \"https://www.ssllabs.com:10303/\") (from (empty))
Failing for the past 5 builds (Since #1 )
Took 1 sec.
Stacktrace

MESSAGE:
System.Net.Http.HttpRequestException : An error occurred while sending the request.\n---- System.Net.Http.CurlException : SSL connect error
+++++++++++++++++++
STACK TRACE:
at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at System.Net.Http.HttpClient.<FinishSendAsync>d__58.MoveNext() in /mnt/resource/j/workspace/dotnet_corefx/dev_apple_crypto/outerloop_centos7.1_release/src/System.Net.Http/src/System/Net/Http/HttpClient.cs:line 380 --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at System.Net.Http.Functional.Tests.HttpClientHandler_SslProtocols_Test.<GetAsync_SupportedSSLVersion_Succeeds>d__6.MoveNext() in /mnt/resource/j/workspace/dotnet_corefx/dev_apple_crypto/outerloop_centos7.1_release/src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.SslProtocols.cs:line 127 --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) ----- Inner Stack Trace ----- at System.Net.Http.CurlHandler.ThrowIfCURLEError(CURLcode error) in /mnt/resource/j/workspace/dotnet_corefx/dev_apple_crypto/outerloop_centos7.1_release/src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs:line 643 at System.Net.Http.CurlHandler.MultiAgent.FinishRequest(StrongToWeakReference`1 easyWrapper, CURLcode messageResult) in /mnt/resource/j/workspace/dotnet_corefx/dev_apple_crypto/outerloop_centos7.1_release/src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.MultiAgent.cs:line 857
````

**Comment by Priya91** [comment]
@steveharter Is this issue tracking test failure that's fixed with #15368

**Comment by Priya91** [comment]
this is fixed. closing.
